pages/auth.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import axios from 'axios'
+import fetch from 'isomorphic-unfetch'
+
+export default class extends React.Component {
+  static async getInitialProps({ req, res }) {
+    const authData = await fetch(
+      (req ? `http://${req.headers.host}` : '') + '/api/authdetails'
+    )
+    const data = await authData.json()
+
+    const code = req.url.match('=(.*)')[1]
+    data.code = code
+
+    const getRefreshToken = async () => {
+      return new Promise((resolve, reject) => {
+        axios({
```

Comment:
Why are we using 3 different libraries for the exact same thing :)

Reply:
becoz i wanna,